core/src/utils/serde/mod.rs

```diff
@@ -55,6 +57,19 @@ impl ByteWriter {
         self.0.extend(val);
     }
 
+    pub fn write_digest(&mut self, digest: &Digest) {
+        // `ElementHasher` is not exposed by winter_crypto API so we must implement manually here
+        digest.write_into(&mut self.0);
+    }
+
+    pub fn write_digests(&mut self, digests: &[Digest]) -> Result<(), SerializationError> {
+        // iterate the elements, using the winter crypto serialization implementation to write the
+        // bytes
+        self.write_len(digests.len())?;
```

Comment:
Since this is now in core, I'm not sure we should assume that length should be encoded as `u16`. It might make sense to get rid of `write_len()` and functions which rely on it and handle length more manually.

In general, it may make sense to reduce the interface surface and maybe get rid of it entirely in favor of the interface exposed by Winterfell - but I think this would require further analysis.

Reply:
Ack! Resolving this discussion to keep the topic centralized on the previous one